vars/buildPlugin.groovy

```diff
@@ -1,43 +1,70 @@
 import hudson.plugins.throttleconcurrents.ThrottleJobProperty;
+import com.cwctravel.hudson.plugins.extended_choice_parameter.ExtendedChoiceParameterDefinition;
 
 /**
  * Simple wrapper step for building a plugin
  */
 def call(Map addonParams = [:])
 {
-	properties([
-		buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '5')),
-		disableConcurrentBuilds(),
-		disableResume(),
-		durabilityHint('PERFORMANCE_OPTIMIZED'),
-		pipelineTriggers(env.BRANCH_NAME == 'master' ? [cron('@weekly')] : []),
-		[$class: 'RebuildSettings', autoRebuild: false, rebuildDisabled: true],
-		[$class: 'ThrottleJobProperty', categories: [], limitOneJobWithMatchingParams: false, maxConcurrentPerNode: 0, maxConcurrentTotal: 1, paramsToUseForLimit: '', throttleEnabled: true, throttleOption: 'category']
-	])
 
 	def PLATFORMS_VALID = [
 		'android-armv7': 'android',
 		'android-aarch64': 'android-arm64-v8a',
 		'ios-armv7': 'ios',
 		'ios-aarch64': 'ios',
 		'osx-x86_64': 'osx64',
+		'ubuntu-ppa': 'linux',
 		'windows-i686': 'windows/win32',
 		'windows-x86_64': 'windows/x64'
 	]
 	def PLATFORMS_DEPLOY = [
 		'android-armv7',
 		'android-aarch64',
 		'osx-x86_64',
+		'ubuntu-ppa',
 		'windows-i686',
 		'windows-x86_64'
 	]
+	def UBUNTU_DISTS = [
+		'cosmic',
+		'bionic',
+		'xenial'
+	]
 	def VERSIONS_VALID = [
 		'master': 'leia',
 		'Leia': 'leia'
 	]
+	def PPAS_VALID = [
+		'nightly': 'ppa:team-xbmc/xbmc-nightly',
+		'unstable': 'ppa:team-xbmc/unstable',
+		'stable': 'ppa:team-xbmc/ppa',
+		'wsnipex-test': 'ppa:wsnipex/xbmc-addons-test'
+	]
+	def PPA_VERSION_MAP = [
+		'master': 'nightly',
+		'Leia': 'unstable'
+	]
+
+	properties([
+		buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '5')),
+		disableConcurrentBuilds(),
+		disableResume(),
+		durabilityHint('PERFORMANCE_OPTIMIZED'),
+		pipelineTriggers(env.BRANCH_NAME == 'master' ? [cron('@weekly')] : []),
+		[$class: 'RebuildSettings', autoRebuild: false, rebuildDisabled: true],
+		[$class: 'ThrottleJobProperty', categories: [], limitOneJobWithMatchingParams: false, maxConcurrentPerNode: 0, maxConcurrentTotal: 1, paramsToUseForLimit: '', throttleEnabled: true, throttleOption: 'category'],
+		parameters([
+			extendedChoice('deployPlatforms', PLATFORMS_DEPLOY.join(','), PLATFORMS_DEPLOY.join(','), 'Platforms to deploy, deploy param from Jenkinsfile is always respected'),
+			extendedChoice('PPA', PPAS_VALID.keySet().join(',')+',auto', 'auto', 'PPA to use'),
+			extendedChoice('dists', UBUNTU_DISTS.join(','), UBUNTU_DISTS.join(','), 'Ubuntu version to build for'),
+			string(defaultValue: '1', description: 'debian package revision tag', name: 'TAGREV', trim: true),
+			booleanParam(defaultValue: false, description: 'Force upload to PPA', name: 'force_ppa_upload')
+		])
```

Comment:
Actually the parameters should be only shown if `env.TAG_NAME != null`

Reply:
would you still be able to rebuild in this case?

Reply:
Yes if you currently click `Build Now` in jenkins it just instantly start the rebuild.

Reply:
I don't think, this is what I want. And I have no idea how I should achieve that. I only got it to show no options, it still says "This build requires parameters:" and the build button.

Reply:
Then just keep it.